sklearn/decomposition/tests/test_pca.py

```diff
@@ -128,16 +130,20 @@ def test_pca_explained_variance_equivalence_solver(svd_solver):
 
 
 @pytest.mark.parametrize(
-    "X",
+    "make_X",
     [
-        np.random.RandomState(0).randn(100, 80),
-        datasets.make_classification(100, 80, n_informative=78, random_state=0)[0],
+        lambda seed: np.random.RandomState(seed).randn(100, 60),
+        lambda seed: datasets.make_classification(
+            100, 80, n_informative=78, random_state=seed
+        )[0],
     ],
     ids=["random-data", "correlated-data"],
 )
 @pytest.mark.parametrize("svd_solver", PCA_SOLVERS)
-def test_pca_explained_variance_empirical(X, svd_solver):
-    pca = PCA(n_components=2, svd_solver=svd_solver, random_state=0)
+def test_pca_explained_variance_empirical(make_X, svd_solver, global_random_seed):
+    # parametrized factory make_X and global_random_seed determine dataset X
+    X = make_X(global_random_seed)
```

Comment:
Instead of directly using `X` in `pytest.mark.parametrize`, I use the factory function `make_X` to inject `global_random_seed` during dataset creation. This is done, because pytest only resolves fixtures within the test function, but not in the decorator.